src/components/Layouts/components/Notification/Notification.tsx

```diff
@@ -0,0 +1,106 @@
+import classNames from 'classnames/bind'
+import styles from './Notification.module.scss'
+
+import Tippy from '@tippyjs/react/headless'
+
+import { IoNotifications } from 'react-icons/io5'
+import PopperWrapper from '../Popper'
+import Button from '~/components/Button'
+import ItemNotification from './ItemNotification'
+import { useEffect, useState } from 'react'
+import { NotificationInterface } from '~/interfaces/notification'
+import { getNotificationByUserIdAPI } from '~/apis/notification'
+import { useSelector } from 'react-redux'
+import { authSelector } from '~/redux/auth/authSelectors'
+const cx = classNames.bind(styles)
+
+const Notification = () => {
+  const [notifications, setNotifications] = useState<NotificationInterface[] | null>(null)
+  const { userInfo } = useSelector(authSelector)
+  const { socket } = useSelector((state: any) => state.socket)
+
+  const fetchNotificationsApi = async () => {
+    try {
+      const response = await getNotificationByUserIdAPI()
+      if (response.statusCode === 200 && response.data.length > 0) {
+        setNotifications(response.data)
+      }
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  useEffect(() => {
+    fetchNotificationsApi()
+
+    return () => {
+      socket?.off('newNotification')
+    }
+  }, [socket])
+  useEffect(() => {
+    if (socket) {
+      socket.on('newNotification', (notification: any) => {
+        setNotifications((prev) => {
+          return [notification, ...(prev || [])]
+        })
+      })
+    }
+    return () => {
+      socket?.off('newNotification')
+    }
+  }, [socket])
+
+  const markAllAsRead = () => {
```

Comment:
Suggest to  use the `useCallback` hook and move this function to the top, same level with the `fetchNotificationsApi`